lib/flipper/api/v1/decorators/feature.rb

```diff
@@ -0,0 +1,33 @@
+require 'delegate'
+require 'flipper/api/v1/decorators/gate'
+
+module Flipper
+  module Api
+    module V1
+      module Decorators
+        class Feature < SimpleDelegator
+
+          # Public: The feature being decorated.
+          alias_method :feature, :__getobj__
+          
+          # Public: Returns instance as hash that is ready to be json dumped.
+          def as_json
+            {
+              'id' => name.to_s,
+              'state' => state.to_s,
+              'gates' => gates.map { |gate|
+                Decorators::Gate.new(gate, gate_values[gate.key]).as_json
```

Comment:
This will cause gate_values to be called for each gate which means an N+1 call against the adapter. That is why [I assign it to a temp var in the UI](https://github.com/jnunemaker/flipper/blob/c25930a6e176d535ce9b21faf4cb5b6c34b098d0/lib/flipper/ui/decorators/feature.rb#L21-L29) and then re-use the temp var.